crates/types/src/serde.rs

```diff
@@ -163,3 +163,177 @@ fn extract_bits(info_word: &BigUint) -> (bool, u64, u64) {
 
     (class_flag, new_nonce, num_changes)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    
+    use std::fs;
+    use num_bigint::BigUint;
+    use std::{result, str::FromStr};
+    use rstest::rstest;
+    use crate::state_diffs::{ClassDeclaration, ContractUpdate, DataJson, StorageUpdate};
+
+    #[rstest]
+    #[case( "18446744073709551617",false, 1, 1)]
+    #[case("18446744073709551616",false, 1, 0)]
+    #[case("6",false, 0, 6)]
+    #[case("340282366920938463481821351505477763072", true, 1, 0)]
+    #[case("0", false, 0, 0)]
+    #[case("340282366920938463463374607431768211455", false, u64::MAX, u64::MAX)]
+    #[case("680564733841876926926749214863536422911", true, u64::MAX, u64::MAX)]
+    #[case("340282366920938486226656794389354915599", true, 1234, 9999)]
```

Comment:
let's also mention the hex values here, it's easier to read the test case that way